daml-foundations/daml-ghc/daml-prim-src/GHC/Base.daml

```diff
@@ -57,12 +57,23 @@ infixr 9 .
 const                   : a -> b -> a
 const x _               =  x
 
+-- | A `Functor` is a typeclass for things that can be mapped over (using
+-- its `fmap` function. Examples include `Maybe` and `[]`.
+
+-- For a detailed explanation, see 
+-- http://learnyouahaskell.com/making-our-own-types-and-typeclasses#the-functor-typeclass
+
 class Functor f where
+    -- | `fmap` must take a function from type `a` to type `b`, and a 
+    -- `Functor` of type `a`, and return a `Functor` of type `b`.
```

Comment:
I would probably slightly rephrase this as something like:
`fmap` takes a function of type `a -> b` and turns it onto a function of type `f a -> f a`, where `f` is the type which is an instance of `Functor`.